opacclient/opacapp/src/main/java/de/geeksfactory/opacclient/frontend/StarredFragment.java

```diff
@@ -256,6 +290,166 @@ protected void export() {
                 getResources().getString(R.string.share)));
     }
 
+    public void exportToStorage() {
+        Intent intent = null;
+        //android 4.4+; use Storage Access Framework
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
+            intent = new Intent(Intent.ACTION_CREATE_DOCUMENT);
+            intent.addCategory(Intent.CATEGORY_OPENABLE);
+            // Create a file with the requested MIME type.
+            intent.setType("application/json");
+            intent.putExtra(Intent.EXTRA_TITLE, "share.json");
+            startActivityForResult(intent, REQUEST_CODE_EXPORT);
+        } else {        // <android 4.4; share raw json file
+            if(Environment.getExternalStorageState()==Environment.MEDIA_MOUNTED) {
+                File dest = new File(getActivity().getExternalCacheDir(),"starred_export.json");
```

Comment:
The problem is, that we then need a new permission, only because of users on Android <4.4 and its especially those users where new permissions break auto-updates so users won't update the app :\